t_sparse_list.py

```diff
@@ -300,5 +300,18 @@ def test_remove_default_value_does_nothing(self):
         sl.remove(None)
         self.assertEquals([None, None, None, None], sl)
 
+    def test_set_slice(self):
+        sl = sparse_list.SparseList(4, None)
+        # Only the first two should be assigned.
+        sl[0:2] = [1, 2, 3]
+        self.assertEquals([1, 2, None, None], sl)
+        # Like a normal list, when we assign past the end, it appends.
+        sl[100:] = [4, 5]
+        self.assertEquals([1, 2, None, None, 4, 5], sl)
+        self.assertEquals(len(sl), 6)
```

Comment:
I find this behaviour surprising.

Given the case:

```python
>>> from sparse_list import SparseList
>>> sl = SparseList(0)
>>> sl
[]
>>> sl[100] = 'test'
>>> len(sl)
101
>>> sl
[None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, test]
```

I would expect `sl[100:] = ` to start inserting at index 100.